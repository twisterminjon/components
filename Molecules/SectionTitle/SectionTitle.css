.sectiontitle-wrap {
  display: flex;
  align-items: baseline;
  overflow: hidden;
}

.sectiontitle-text {
  color: var(--brandcolor);
}

.sectiontitle-rule {
  border: 0;
  height: 2px;
  background: var(--brandcolor);
}
