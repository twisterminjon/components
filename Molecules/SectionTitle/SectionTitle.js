import React, { Component } from 'react'
import PropTypes from 'prop-types'
import TitleText from '../../Atoms/TitleText/TitleText'
import HintText from '../../Atoms/HintText/HintText'
import TextButton from '../../Atoms/TextButton/TextButton'
import './SectionTitle.css'

export default class SectionTitle extends Component {
  static propTypes = {
    /** Text for Title */
    title: PropTypes.string.isRequired,
    /** Text for hint */
    hint: PropTypes.string,
    /** Label for button */
    label: PropTypes.string,
    /** Function called on button click */
    onButtonClick: PropTypes.func,
    /** ClassName for the wrapper */
    className: PropTypes.string,
    // FIXME: buttonClassName doesn't work, see TextButton for the reason :(
    /** Classname for the button wrapper */
    buttonClassName: PropTypes.string,
    /** Style for the wrapper */
    style: PropTypes.object,
  }
  static defaultProps = {
    hint: '',
    label: '',
    className: '',
    buttonClassName: '',
    style: {},
    onButtonClick: () => {},
  }

  render() {
    const { title, className, hint, label, onButtonClick, style } = this.props

    const button = label ? (
      <TextButton
        style={{ marginLeft: 'auto', marginRight: 2 }}
        onClick={onButtonClick}
        data-testid="section-button"
        type="button"
        content={label}
      />
    ) : null

    return (
      <div className={className} style={style}>
        <div className="sectiontitle-wrap">
          <TitleText title={title} className="sectiontitle-text" />
          <HintText hint={hint} />
          {button}
        </div>
        <hr className="sectiontitle-rule" />
      </div>
    )
  }
}
