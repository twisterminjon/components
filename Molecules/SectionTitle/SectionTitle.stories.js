import React from 'react'

import { storiesOf } from '@storybook/react'
import centered from '@storybook/addon-centered/react'
import { withKnobs, text } from '@storybook/addon-knobs'

import SectionTitle from './SectionTitle'

const stories = storiesOf('Provider/v1/Molecules', module)
stories.addDecorator(withKnobs)

stories.addDecorator(centered).add('SectionTitle', () => {
  const textLabel = 'title'
  const textDefaultValue = 'The stash'
  const textValue = text(textLabel, textDefaultValue)

  const hintLabel = 'hint'
  const hintDefaultValue = 'Psst..here is a secret ;)'
  const hintValue = text(hintLabel, hintDefaultValue)

  const buttonLabel = 'buttonLabel'
  const buttonDefaultValue = 'Add'
  const buttonValue = text(buttonLabel, buttonDefaultValue)

  return (
    <div style={{ width: 400 }}>
      <SectionTitle
        title={textValue}
        hint={hintValue}
        label={buttonValue}
        onButtonClick={() => {
          alert('Clicked Add')
        }}
      />
    </div>
  )
})
