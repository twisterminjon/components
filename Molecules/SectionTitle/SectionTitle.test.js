import React from 'react'
import SectionTitle from './SectionTitle'

describe('SectionTitle', () => {
  const mockFun = jest.fn()
  const title = 'Help me'
  const hint = 'this is spinal tap'
  const buttonLabel = 'ADD'

  it('matches the snapshot', () => {
    const wrapper = window.shallow(<SectionTitle title={title} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('contains the correct hint text', () => {
    const wrapper = window.mount(<SectionTitle title={title} hint={hint} />)

    expect(wrapper.find(`[hint="${hint}"]`)).toBeTruthy()
  })

  it('shows the button when a label is provided', () => {
    const wrapper = window.mount(<SectionTitle title={title} hint={hint} label={buttonLabel} />)

    expect(wrapper.find('[data-testid="section-button"]')).not.toBeNull()
  })

  it('contains the correct title text', () => {
    const wrapper = window.mount(<SectionTitle title={title} hint={hint} />)

    expect(wrapper.find('h3').text()).toEqual(title)
  })

  it('calls the onButtonClick function', () => {
    const wrapper = window.shallow(
      <SectionTitle title={title} hint={hint} label={buttonLabel} onButtonClick={mockFun} />
    )

    wrapper.find('[data-testid="section-button"]').simulate('click')

    expect(mockFun).toHaveBeenCalled()
  })
})
