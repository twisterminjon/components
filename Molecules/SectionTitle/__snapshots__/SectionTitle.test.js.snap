// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionTitle matches the snapshot 1`] = `
<div
  className=""
  style={Object {}}
>
  <div
    className="sectiontitle-wrap"
  >
    <TitleText
      className="sectiontitle-text"
      style={Object {}}
      title="Help me"
    />
    <HintText
      hint=""
      rightAlign={false}
    />
  </div>
  <hr
    className="sectiontitle-rule"
  />
</div>
`;
